### DISCUSSION TITLE
Strange rule name `known_macro_invocation_repeat2`, and why parser generation fails?

### DISCUSSION THREAD

**Discussion by svblsvljb** [tree-sitter/tree-sitter#2028] [Q&A] [answered]
I have the [following](https://aruthecrsov.com/files/for-tree-sitter-discussion/grammar.js) grammar forked from [tree-sitter-rust](https://github.com/tree-sitter/tree-sitter-rust), which in turn `require`s [generated.js](https://aruthecrsov.com/files/for-tree-sitter-discussion/generated.js) (I am generating this file using an external program).

`tree-sitter generate` fails with this output:
```
The rule `known_macro_invocation_repeat2` matches the empty string.

Tree-sitter does not support syntactic rules that match the empty string
unless they are used only as the grammar's start rule.
```

What is `known_macro_invocation_repeat2`? There is only `known_macro_invocation` rule in generated.js. How does `known_macro_invocation_repeat2` relate to `known_macro_invocation`?

**Comment by svblsvljb** [comment]
While `tree-sitter generate` fails, both `tree-sitter build-wasm` and `tree-sitter parse <SOMERUSTFILE>` works...

**Comment by ahelwer** [comment, marked as answer]
If you `repeat()` a rule then tree-sitter unrolls it during code generation, and if it encounters an error you'll see the unrolled rule name.
